src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LambdaParameterNameCheck.java

```diff
@@ -113,10 +113,11 @@ public LambdaParameterNameCheck() {
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST parametersNode = ast.findFirstToken(TokenTypes.PARAMETERS);
-        if (parametersNode == null) {
+        if (parametersNode == null
+                && ast.getParent().getType() != TokenTypes.SWITCH_RULE) {
```

Comment:
Please split this `if` into two and revert `else if`. The condition will be easy to read. Or, the whole condition can be inverted:
```java
        final DetailAST parametersNode = ast.findFirstToken(TokenTypes.PARAMETERS);
        if (parametersNode != null) {
            TokenUtil.forEachChild(parametersNode, TokenTypes.PARAMETER_DEF, super::visitToken);
        }
        else if (ast.getParent().getType() != TokenTypes.SWITCH_RULE) {
            super.visitToken(ast);
        }
```

Reply:
Causes PMD violation: ` PMD Failure: com.puppycrawl.tools.checkstyle.checks.naming.LambdaParameterNameCheck:117 Rule:ConfusingTernary Priority:3 Avoid if (x != y) ..; else ..;.`

I've worked around it with a method call, `Objects.nonNull(ast.findFirstToken(TokenTypes.PARAMETERS))`